app/code/Magento/Elasticsearch/SearchAdapter/Query/Builder/MatchQuery.php

```diff
@@ -125,30 +129,31 @@ protected function prepareQuery($queryValue, $conditionType)
      * @param array $queryValue
      * @return array
      */
-    protected function buildQueries(array $matches, array $queryValue)
+    private function buildQueries(array $matches, array $queryValue): array
```

Comment:
Please revert

Reply:
Here is the reason why it's needed: https://github.com/magento/magento2/pull/33632#pullrequestreview-721876346